File: app/src/main/java/com/qiaoqiao/app/AppModule.java
package com.qiaoqiao.app;


import android.content.Context;
import android.support.annotation.NonNull;

import java.io.IOException;
import java.util.Properties;

import dagger.Module;
import dagger.Provides;

@Module
public final class AppModule {
	private @NonNull App mApp;


	AppModule(@NonNull App app) {
		mApp = app;
	}

	@Provides
	Context provideApp() {
		return mApp;
	}

	@Provides
	Key provideKey(@NonNull Context cxt) {
		String key = null;
		Properties prop = new Properties();
		try {
			prop.load(cxt.getClassLoader()
			             .getResourceAsStream("key.properties"));
			key = prop.getProperty("apikey");
		} catch (IOException e) {
			e.printStackTrace();
		}
		return new Key(key);
	}
}

File: app/src/main/java/com/qiaoqiao/app/AppModule.kt
package com.qiaoqiao.app

import android.content.Context
import dagger.Module
import dagger.Provides
import java.util.*

@Module
class AppModule internal constructor(private val mApp: App) {

    @Provides
    internal fun provideApp(): Context {
        return mApp
    }

    @Provides
    internal fun provideKey(cxt: Context): Key {
        with(Properties()) {
            load(cxt.classLoader.getResourceAsStream("key.properties"))
            val key = getProperty("apikey")
            key?.let {
                return Key(it)
            }
        }
        throw NullPointerException("The Api-key can't be found in file key.properties .")
    }
}
